src/sinks/loki/service.rs

```diff
@@ -46,6 +66,7 @@ impl DriverResponse for LokiResponse {
     }
 }
 
+#[derive(Clone)]
 pub struct LokiRequest {
```

Comment:
Hmm, why does `LokiRequest` require `Clone`? That would seem to have performance implications, no?

Reply:
This is a long-standing issue with the retry layer needing the request to be `Clone`.

Ideally, we would build requests using `Bytes` and/or `Arc<String>` or some other cheaply-cloned shared pointer type.